### ISSUE TITLE
Yolo boxes format overflowing image and thowing an exception

### ISSUE THREAD

**Issue by Trud09** [albumentations-team/albumentations#848]
## 🐛 Bug

I have some bounding boxes in yolo format that are off the image.

## To Reproduce

1. Have boxes outside image in yolo format and try to augment

\lib\site-packages\albumentations\augmentations\bbox_utils.py", line 330, in check_bbox
    "to be in the range [0.0, 1.0], got {value}.".format(bbox=bbox, name=name, value=value)
ValueError: Expected x_max for bbox (0.9967206790123457, 0.5379050925925926, 1.0001929012345678, 0.5529513888888888, 0) to be in the range [0.0, 1.0], got 1.0001929012345678.

## Expected behavior

I would prefer it to clip the boxes, or at least give me the option to do so.

## Environment

 - Albumentations version (e.g., 0.1.8): 0.5.2
 - Python version (e.g., 3.7): 3.6
 - OS (e.g., Linux): Windows
 - How you installed albumentations (`conda`, `pip`, source): pip
 - Any other relevant information:

**Comment by ternaus** [comment]
It is not a bug. It is a feature.

It is much easier to pre-process labels once before model training, rather than catching the challenging corner cases.

Feel free to add one line that clips the bounding boxes to your dataloader :)

**Comment by Trud09** [comment]
Actually upon further investigation, there is a bug somewhere.
When I convert the boxes on my own from 'yolo' to 'albumentations' format, it caps at 1, and does not throw an exception. So somewhere it is adding a small amount, maybe in rounding somewhere, causing it to be greater than 1.

**Comment by ternaus** [comment]
Could you please provide an example that reproduces the behavior?

**Comment by Trud09** [comment]
The image might be proprietary, but here is one box causing the issue:

Yolo Format:
[0.99662423, 0.7520255 , 0.00675154, 0.01446759]

My albumentations conversion:
[0.99324846, 0.7447917, 1.0, 0.7592593]

Exception for this box:
Exception has occurred: ValueError
Expected x_max for bbox (0.9934413580246914, 0.7450810185185185, 1.0001929012345678, 0.7592592592592593, 0) to be in the range [0.0, 1.0], got 1.0001929012345678.

My conversion code:
def yolo_to_norm_voc(bboxes):
    x_mean, y_mean, w, h = bboxes[:, 0], bboxes[:, 1], bboxes[:, 2], bboxes[:, 3]

    x_min = x_mean - (w/2)
    y_min = y_mean - (h/2)

    x_min, x_max, y_min, y_max = x_min, x_min + w, y_min, y_min + h
    return np.stack([x_min, y_min, x_max, y_max], 1)

**Comment by sang-yc** [comment]
Before data augmentation, the length and width of GT bboxes are subtracted by one pixel, which has little effect on the label bounding box, but it can prevent the overflow of values and prevent the generation of labels greater than 1.

labels format: yolo
image.shape[0]: width of image
image.shape[1]: length of image
bboxes: [[x, y, w, h, class], [x, y, w, h, class], ...]

```python
for i in range(len(bboxes)):
    bboxes[i][2] = np.abs(bboxes[i][2] - 0.5 / image.shape[0])
    bboxes[i][3] = np.abs(bboxes[i][3] - 0.5 / image.shape[1])
```
To further reduce the impact of processing, I only subtracted 0.5 pixels from the code.

**Comment by merryHunter** [comment]
Same Value error goes for 'coco' boxes. Indeed current workaround is to clip boxes or np.round after the transformations, but not sure that's the intended behavior. I believe most likely there are some problems deep under the hood with floats.

**Comment by Dipet** [comment]
Should be fixed by #924
